packages/react-ui/src/widgets/legend/LegendWrapper.js

```diff
@@ -79,14 +83,15 @@ export default function LegendWrapper({
         layerOptionsEnabled={showOpacityControl || layerOptions.length > 0}
         onToggleLayerOptions={handleToggleLayerOptions}
         isLayerOptionsExpanded={isLayerOptionsExpanded}
+        locale={locale}
       />
       {hasChildren && !!children && (
         <Collapse ref={wrapper} in={expanded} timeout='auto' unmountOnExit>
           <Box sx={{ pt: '2px', pl: 3, pr: 2, pb: 2 }}>
             <Grid container direction='column' spacing={1}>
               {attr && (
                 <Typography xs mb={1} variant='caption'>
-                  By {attr}
+                  {intl.formatMessage({ id: 'c4r.widgets.legend.by' })} {attr}
```

Comment:
you don't know grammar of other languages, create message `by {attr}` and pass `attr` in `values`

Reply:
Done